native/acter/src/api/stream.rs

```diff
@@ -80,19 +94,195 @@ impl TimelineStream {
         matches!(self.room.state(), RoomState::Joined)
     }
 
-    pub async fn edit(&self, new_msg: String, original_event_id: String) -> Result<bool> {
+    // for integration test, not api
+    pub async fn event(&self, event_id: OwnedEventId) -> Result<EventTimelineItem> {
+        let timeline = self.timeline.clone();
+
+        RUNTIME
+            .spawn(async move {
+                let item = timeline
+                    .item_by_event_id(&event_id)
+                    .await
+                    .context("The event not found")?;
+                Ok(item)
+            })
+            .await?
+    }
+
+    pub async fn send_plain_message(&self, message: String) -> Result<bool> {
```

Comment:
I wonder if we want to rethink this entire API. In particular I wonder if might want to have a `MessageBuilder` that the user can just use directly to construct their message and then have a singular API can consume via a `timeline.send(message.into()).await`. Would make it easier to add features than having to add a new function every time ...